task7/class.js

```diff
@@ -0,0 +1,307 @@
+   	createdAt: new Date('2020-03-23T23:00:00'),
+   	author: 'Да да я',
+   	photoLink: 'images/1.png',
+   	hashTags: ['#gelo','#ni'],
+   	likes: ['max', 'Иванов Иван']
+	},
+	{
+		id: '19',
+   	description: 'фЫврш гффы фы тв ргц фыв',
+   	createdAt: new Date('2020-03-24T23:00:00'),
+   	author: 'Да да я',
+   	photoLink: 'images/1.png',
+   	hashTags: ['#gelo','#ni'],
+   	likes: ['max', 'Иванов Иван']
+	},
+	{
+		id: '20',
+   	description: 'фЫврш гффы фы тв ргц фыв',
+   	createdAt: new Date('2020-03-25T23:00:00'),
+   	author: 'Да да я',
+   	photoLink: 'images/1.png',
+   	hashTags: ['#gelo','#ni'],
+   	likes: ['max', 'Иванов Иван']	
+	},
+];
+class PostList{
+	constructor(posts){
+		this._posts = posts;
+	}
+	_comp(a, b){
+		return a.createdAt - b.createdAt;
+	}
+	getPage(skip = 0, top = 10, filterConfig ={}) {
+			let new_posts = [];
+			let filterKeys = Object.keys(filterConfig);
+			let filt;
+			if(filterKeys.length != 0)
+			{
+				filt = this._posts.filter(post => {
+					if(filterKeys[0] != 'hashTags')
+						return post[filterKeys[0]] == filterConfig[filterKeys[0]]
+					for(let tag of filterConfig.hashTags)
+						if(post.hashTags.indexOf(tag) >= 0)
+							return true;
+					return false;
+				 });
+			}
+			else
+				filt = this._posts;
+			filt.sort(this.comp);
+			for (let i = skip; i < Math.min(skip + top, filt.length); i++) {
+				new_posts.push(filt[i]);
+			}
+			return new_posts;
+		}
+	get(id = '1'){
+		for(let post of this._posts)
+			if(post.id == id)
+				return post;
+		return null;
+	}
+	static _validate(post){
+			if(!post.hasOwnProperty('id') || typeof(post.id)!='string')
+				return false;
+			if(!post.hasOwnProperty('description') || typeof(post.description)!='string' || post.description.length > 200)
+				return false;
+			if(!post.hasOwnProperty('createdAt') || !((post.createdAt) instanceof Date) ||post.createdAt=='Invalid Date')
+				return false;
+			if(!post.hasOwnProperty('author') || typeof(post.author)!='string' || post.author.length==0)
+				return false;
+			if(!post.hasOwnProperty('likes') || !Array.isArray(post.likes))
+				return false;
+			return true;
+	}
+	add(post){
+			if(PostList._validate(post)){
+				this._posts.push(post);
+				return true;
+			}
+			return false;
+	}
+	edit(id,post){
+			let ind = 0;
+			for(let i = 0; i < this._posts.length;i++)
+				if(this._posts[i].id == id)
+					{
+						ind = i;
+						break;
+					}
+			let editPost = this._posts[ind];
+			for(let key in post){
+				if(key != 'id' && key != 'createdAt' && key != 'author' && key!='likes')
+					editPost[key] = post[key];
+			}
+			if(!PostList._validate(editPost))
+				return false;
+			this._posts[ind] = editPost;
+			return true;
+	}
+	remove(id){
+			for(let i = 0; i < this._posts.length; i++){
+				if(this._posts[i].id == id)
+				{
+					this._posts.splice(i,1);
+					return true;
+				}
+			}
+			return false;
+		}
+	addAll(posts){
+		let wrong = [];
+		for(let post of posts)
```

Comment:
It's better to use forEach method https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach